test/integration/auth/auth_test.go

```diff
@@ -553,6 +557,17 @@ func TestAuthModeAlwaysDeny(t *testing.T) {
 	// Set up an API server
 	controlPlaneConfig := framework.NewIntegrationTestControlPlaneConfig()
 	controlPlaneConfig.GenericConfig.Authorization.Authorizer = authorizerfactory.NewAlwaysDenyAuthorizer()
+	privilegedLoopbackToken := uuid.New().String()
+	// wrap any available authorizer
+	tokens := make(map[string]*user.DefaultInfo)
+	tokens[privilegedLoopbackToken] = &user.DefaultInfo{
+		Name:   user.APIServerUser,
+		UID:    uuid.New().String(),
+		Groups: []string{user.SystemPrivilegedGroup},
+	}
+	tokenAuthenticator := authenticatorfactory.NewFromTokens(tokens, controlPlaneConfig.GenericConfig.Authentication.APIAudiences)
+	// TestAuthModeAlwaysDeny need an unprivileged user, specifically setting that up
+	controlPlaneConfig.GenericConfig.Authentication.Authenticator = authenticatorunion.New(tokenAuthenticator, authauthenticator.RequestFunc(framework.AlwaysEmpty))
```

Comment:
drop the custom authenticator here and set up the transport below to use the unprivileged user token:

```go
import (
...
  resttransport k8s.io/client-go/transport
...
)
...

transport := resttransport.NewBearerAuthRoundTripper(framework.UnprivilegedUserToken, http.DefaultTransport)
```

Reply:
done